poh/src/poh_recorder.rs

```diff
@@ -449,16 +475,53 @@ impl PohRecorder {
         self.start_bank = reset_bank;
         self.tick_height = (self.start_slot() + 1) * self.ticks_per_slot;
         self.start_tick_height = self.tick_height + 1;
+        self.leader_tick_heights = LeaderTickHeights::compute_next_tick_heights(
+            next_leader_slots,
+            self.ticks_per_slot,
+            self.tick_height,
+        );
+    }
+
+    // Report how much time has elapsed before leader ticks being reached and the bank getting set
+    // to track the delay between a leader being able to produce block and the bank getting created.
+    fn report_leader_delay_metrics(&self, current_leader_slot: Slot) {
+        if let Some(leader_tick_heights) = self.leader_tick_heights.as_ref() {
+            let first_leader_slot =
+                self.slot_for_tick_height(leader_tick_heights.first_tick_height);
+            if current_leader_slot == first_leader_slot {
+                if let Some(first_tick_reached_at) =
+                    leader_tick_heights.first_tick_reached_at.as_ref()
+                {
+                    datapoint_info!(
+                        "poh-recorder-set-bank-stats",
+                        ("slot", current_leader_slot, i64),
+                        (
+                            "leader_first_tick_elapsed_us",
+                            first_tick_reached_at.elapsed().as_micros(),
+                            i64
+                        ),
+                    );
+                }
 
-        let (leader_first_tick_height_including_grace_ticks, leader_last_tick_height, grace_ticks) =
-            Self::compute_leader_slot_tick_heights(next_leader_slot, self.ticks_per_slot);
-        self.grace_ticks = grace_ticks;
-        self.leader_first_tick_height_including_grace_ticks =
-            leader_first_tick_height_including_grace_ticks;
-        self.leader_last_tick_height = leader_last_tick_height;
+                if let Some(target_tick_reached_at) =
+                    leader_tick_heights.target_tick_reached_at.as_ref()
+                {
+                    datapoint_info!(
+                        "poh-recorder-set-bank-stats",
+                        ("slot", current_leader_slot, i64),
+                        (
+                            "leader_target_tick_elapsed_us",
```

Comment:
nit: prefer to keep the original style of more explicit naming "target_tick_including_grace_ticks_elapsed_us"